unleash/src/main/java/com/silvercar/unleash/repository/JsonToggleParser.kt

```diff
@@ -0,0 +1,23 @@
+package com.silvercar.unleash.repository
+
+import java.io.Reader
+
+internal class JsonToggleParser {
+  private val gsonFactory = GsonFactory
+  fun toJsonString(toggleCollection: ToggleCollection?): String {
+    return gsonFactory.getInstance().toJson(toggleCollection)
+  }
+
+  @Throws(IllegalStateException::class)
+  fun fromJson(reader: Reader?): ToggleCollection {
+    val gson = gsonFactory.getInstance()
+    val toggleCollection = gson.fromJson(reader, ToggleCollection::class.java)
+
+    check(toggleCollection?.features != null) {
```

Comment:
If I'm reading this correctly, shouldn't we see if the features _are_ equal to null? Or does this this mean that the `{ ... }` will be executed as an else?

Reply:
`{...}` is an else.

`fun check(value: Boolean)` // Throws an IllegalStateException if the value is false.